File: DIV2/ki.py
import math

# Función para calcular la distancia entre dos puntos en el espacio tridimensional
def distancia(x1, y1, z1, x2, y2, z2):
    return math.sqrt((x2-x1)**2 + (y2-y1)**2 + (z2-z1)**2)

# Función para calcular el tiempo que tarda el Kamehameha en llegar al planeta Kranus
def tiempo_kamehameha(distancia, velocidad):
    return distancia / velocidad

# Función para calcular el tiempo que tarda Goku en cargar la mitad de su Ki en el planeta Kranus
def tiempo_carga(ki):
    return 0.5 * ki / 10000
# Lectura de datos y procesamiento de cada caso de prueba
while True:
    try:
        x1, y1, z1, x2, y2, z2, velocidad, ki = map(float, input().split())
    except:
        break
    
    dist = distancia(x1, y1, z1, x2, y2, z2)
    print(dist)
    tiempo_k = tiempo_kamehameha(dist, velocidad)
    print(tiempo_k)
    if dist == 0 :
        tiempo_c = 0
    else:
        tiempo_c = tiempo_carga(ki)
    print(tiempo_c)
    if tiempo_k < tiempo_c:
        print("Lanzar kamehameha")
    elif tiempo_k > tiempo_c:
        print("Teleportacion")
    else:
        print("Teleportacion")

File: DIV2/ki.cpp
#include <iostream>
#include <cmath>

using namespace std;

const double EPS = 1e-9;

double distance(double x1, double y1, double z1, double x2, double y2, double z2) {
    double dx = x1 - x2, dy = y1 - y2, dz = z1 - z2;
    return sqrt(dx*dx + dy*dy + dz*dz);
}

int main() {
    double x1, y1, z1, x2, y2, z2, v, k;
    while (cin >> x1 >> y1 >> z1 >> x2 >> y2 >> z2 >> v >> k) {
        double dist = distance(x1, y1, z1, x2, y2, z2);
        double timeKamehameha = dist / v;
        double timeTeleportation = 0.5 * k / 5000; // ki carga fija a 10000 por segundo
        if (timeTeleportation <= timeKamehameha) {
            cout << "Teleportacion\n";
        } else {
            cout << "Lanzar kamehameha\n";
        }
    }
    return 0;
}
